doc/hooks/waf.section.md

```diff
@@ -1,16 +1,16 @@
-# wafHook {#wafhook}
+# waf.hook {#wafhook}
 
 [Waf](https://waf.io) is a Python-based software building system.
 
-In Nixpkgs, `wafHook` overrides the default configure, build, and install phases.
+In Nixpkgs, `waf.hook` overrides the default configure, build, and install phases.
 
-## Variables controlling wafHook
+## Variables controlling waf.hook {#variablesControllingWafHook}
```

Comment:
I think this should land in staging-next to fix the build of the Manual

Reply:
Is this allowed?

Reply:
Sorry, I mean the build of the Nixpkgs Manual is failing on staging-next because this heading doesn't have an id https://github.com/NixOS/nixpkgs/actions/runs/5960199606/job/16167111983

So I think we need to add something like https://github.com/marsam/nixpkgs/commit/df0d6d6c0e210ffafd91f05774b5ab8b3e439605 or land this PR on staging-next

Reply:
Oh, I think I understood now.
What is the path of least resistance? I plan to continue the refactor of this expression at https://github.com/NixOS/nixpkgs/pull/250889 (which includes changes on code outside waf)